### ISSUE TITLE
Stuck getting the slaves to report to the master

### ISSUE THREAD

**Issue by memelet** [twosigma/satellite#44]
I'm stuck and could use a little guidance. 

I have the master runnign on the mesos-masters. The two masters that are not on the mesos master node are following each other (according to the logs). The leader master comes up with

```
2015-11-26 10:40:21,977 INFO  satellite.core [main] - Starting Satellite
2015-11-26 10:40:21,978 INFO  satellite.core [main] - Reading config from file: /etc/satellite/satellite-config.clj
2015-11-26 10:40:22,397 INFO  satellite.riemann.services.leader [clojure-agent-send-off-pool-0] - Starting leader service
2015-11-26 10:40:22,401 INFO  satellite.riemann.services.leader [clojure-agent-send-off-pool-0] - Leader service started
2015-11-26 10:40:22,402 INFO  satellite.riemann.services.whitelist [clojure-agent-send-off-pool-0] - Starting whitelist service
2015-11-26 10:40:22,429 INFO  satellite.riemann.services.http [clojure-agent-send-off-pool-1] - Starting http server on port 5001
2015-11-26 10:40:22,429 INFO  riemann.core [main] - Hyperspace core online
2015-11-26 10:40:22,441 INFO  satellite.riemann.services.curator [clojure-agent-send-off-pool-3] - Starting curator service
2015-11-26 10:40:22,898 INFO  org.apache.curator.framework.imps.CuratorFrameworkImpl [clojure-agent-send-off-pool-3] - Starting
2015-11-26 10:40:22,931 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:zookeeper.version=3.4.6-1569965, built on 02/20/2014 09:09 GMT
2015-11-26 10:40:22,931 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:host.name=s-mesos-master-2
2015-11-26 10:40:22,932 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:java.version=1.8.0_66-internal
2015-11-26 10:40:22,932 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:java.vendor=Oracle Corporation
2015-11-26 10:40:22,932 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:java.home=/usr/lib/jvm/java-8-openjdk-amd64/jre
2015-11-26 10:40:22,932 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:java.class.path=satellite-master.jar
2015-11-26 10:40:22,932 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib/x86_64-linux-gnu/jni:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/jni:/lib:/usr/lib
2015-11-26 10:40:22,932 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:java.io.tmpdir=/tmp
2015-11-26 10:40:22,933 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:java.compiler=<NA>
2015-11-26 10:40:22,933 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:os.name=Linux
2015-11-26 10:40:22,933 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:os.arch=amd64
2015-11-26 10:40:22,933 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:os.version=3.13.0-63-generic
2015-11-26 10:40:22,933 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:user.name=root
2015-11-26 10:40:22,933 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:user.home=/root
2015-11-26 10:40:22,933 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Client environment:user.dir=/opt/satellite/satellite-master
2015-11-26 10:40:22,934 INFO  org.apache.zookeeper.ZooKeeper [clojure-agent-send-off-pool-3] - Initiating client connection, connectString=10.0.137.248:2181 sessionTimeout=180000 watcher=org.apache.curator.ConnectionState@4279f493
2015-11-26 10:40:22,937 INFO  riemann.transport.tcp [clojure-agent-send-off-pool-4] - TCP server 10.0.137.248 5555 online
2015-11-26 10:40:22,944 INFO  satellite.riemann.monitor [clojure-agent-send-off-pool-4] - Tracking follower http://10.0.137.248:5050
2015-11-26 10:40:22,999 INFO  org.apache.zookeeper.ClientCnxn [clojure-agent-send-off-pool-3-SendThread(10.0.137.248:2181)] - Opening socket connection to server 10.0.137.248/10.0.137.248:2181. Will not attempt to authenticate using SASL (unknown error)
2015-11-26 10:40:23,012 INFO  org.apache.zookeeper.ClientCnxn [clojure-agent-send-off-pool-3-SendThread(10.0.137.248:2181)] - Socket connection established to 10.0.137.248/10.0.137.248:2181, initiating session
2015-11-26 10:40:23,026 INFO  org.apache.zookeeper.ClientCnxn [clojure-agent-send-off-pool-3-SendThread(10.0.137.248:2181)] - Session establishment complete on server 10.0.137.248/10.0.137.248:2181, sessionid = 0x251398577df45d4, negotiated timeout = 40000
2015-11-26 10:40:23,031 INFO  org.apache.curator.framework.state.ConnectionStateManager [clojure-agent-send-off-pool-3-EventThread] - State change: CONNECTED
2015-11-26 10:40:24,045 INFO  satellite.riemann.services.curator [clojure-agent-send-off-pool-3] - Curator service started
2015-11-26 10:40:24,089 INFO  satellite.riemann.services.whitelist [clojure-agent-send-off-pool-0] - Whitelist service started
2015-11-26 10:40:24,105 INFO  satellite.riemann.services.http [clojure-agent-send-off-pool-1] - Http server started
2015-11-26 10:40:24,121 INFO  org.eclipse.jetty.server.Server [clojure-agent-send-off-pool-0] - jetty-9.2.z-SNAPSHOT
2015-11-26 10:40:24,150 INFO  org.eclipse.jetty.server.ServerConnector [clojure-agent-send-off-pool-0] - Started ServerConnector@56b79007{HTTP/1.1}{10.0.137.248:5001}
2015-11-26 10:40:24,151 INFO  org.eclipse.jetty.server.Server [clojure-agent-send-off-pool-0] - Started @11995ms
```

All the slaves output:

```
2015-11-26 10:41:06,483 INFO  satellite-slave.core [main] - Starting Satellite-Slave
2015-11-26 10:41:06,484 INFO  satellite-slave.core [main] - Reading config from file: /etc/satellite/slave-config.clj
```

My riemann config is simply (as per the debugging section in the docs)

```
(stream prn)
```

The above is all I ever see the the logs. 

I used the riemann-cli tool to send messages directly from the slave node to the master node, and those DO print out in the master log. But nothing seems to get there from the satellite slaves.

I also ran the slave in --dry-run and it DID emit messages for all checks on the intervals specified in the config.

I'm not sure what else I can do before I fork the repo and start adding log statements.

**Comment by memelet** [comment]
Maybe this has to do with the rest api port vs the riemann port? Are they the same port? I don't see any way to configure the riemann port.

**Comment by memelet** [comment]
Ok, I see. The :satellite is passed to tcp-client, which /can/ take :port. Since there is no mention of the riemann port in the docs, I did not specify one. And since by satellite's are running docker the default riemann port (5555) was not exposed.

**Comment by memelet** [comment]
Still struggling. I can give the slave a completly bogus riemann host/port and it is happy send events into the ether with no errors at all. 

```
2015-11-27 13:16:46,036 DEBUG satellite-slave.core [async-thread-macro-1] - Sent event: {:service "mesos/slave/", :time 1448610406, :description "example test -- number of files/dirs in cwd", :state "ok", :ttl 300, :metric 100.0}
...
```

**Comment by memelet** [comment]
Ok, not sure what I did, but I'm getting events. Could be simply that I was only looking at the log and not stdout for the (streams prn).

Summary of problems from this thread:
- Riemann was using default port 5555, which I did not have exposed via docker nor open in my aws security group. To find about this port I had to trace into the riemann code. It would be nice if the examples for :riemann-tcp-server-options and :satellites included the port. While now it obvious, for a bit it was not clear to me that the master was listening on two different ports (rest-api and riemann).
- I was using (stream prn) vs (streams prn) for a bit there.
